src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php

```diff
@@ -237,17 +241,30 @@ protected function configureOptions(OptionsResolver $resolver): void
 
     private function createTranslations(ProductInterface $product, array $options): void
     {
+        // add translation for each defined locales
         foreach ($this->getLocales() as $localeCode) {
-            $product->setCurrentLocale($localeCode);
-            $product->setFallbackLocale($localeCode);
+            $this->createTranslation($product, $localeCode, $options);
+        }
 
-            $product->setName($options['name']);
-            $product->setSlug($options['slug']);
-            $product->setShortDescription($options['short_description']);
-            $product->setDescription($options['description']);
+        // create or replace with custom translations
+        foreach ($options['translations'] as $localeCode => $translationOptions) {
+            $this->createTranslation($product, $localeCode, $translationOptions);
         }
     }
 
+    private function createTranslation(ProductInterface $product, string $localeCode, array $options = []): void
+    {
+        $options = $this->optionsResolver->resolve($options);
```

Comment:
I've spotted a problem with the current implementation. We shouldn't resolve options again here(that's why you had to add this filtration in the first place).

What should be done, is to remove filtration logic from product attributes, extract translation a logic to separate example factory and delegate work into it(more or less what is happening here). The most tricky part would be to avoid bc-breaks, so we need to keep the current behavior, and use this additional logic only for defined transition.

It should be separate service, with an allowance of injecting null

Reply:
Going back and forth like this is not really doing it for me. Feel free to push the desired changes.

Reply:
Oh, excuse me for pushing back and forth. I didn't want to push you away from these changes. This PR could be a beneficial enhancement for the platform.

For sure, the feedback loop could be faster, and I'm sorry about that. We are doing our best to be daily available on this repository. 

However, to spot this problem and provide you finally (I hope) valuable feedback about this change, I had to fetch your branch, and thanks to your tests, debug it with xdebug. It's not possible to do with every single PR, and that is why I haven't done it in the first place. Due to our rather strict bc policy, we need to be careful what lands in the codebase. The current solution is more like a workaround so that it couldn't be merged.

 As today we've started feature freeze phase, this PR will not be released before 1.8 (September 2020), we have some time to improve this solution. I will ask our PO to add it to the backlog. Maybe it will help this feature to land into the core.

Thanks again for your time and patience. I hope we will have another opportunity to work together on some changes in the future, and it will go smoother than this time.

Reply:
Thank you for your kind words. Since [we became a partner](https://sylius.com/partners/enrise/) I think we could work together in the future for sure.